contracts/MMRStatelessVerification.sol

```diff
@@ -0,0 +1,50 @@
+// "SPDX-License-Identifier: UNLICENSED"
+pragma solidity ^0.7.0;
+
+contract MMRStatelessVerification {
+
+    // Verifies is a parachain header is contained in an MMR leaf
+    function verifyParachainHeaderInLeaf(
+        bytes32 paraHead,
+        bytes32 paraHeadSibling,
+        bytes32 leafParaHeadRoot
+    ) public pure returns (bool) {
+        return leafParaHeadRoot == keccak256(abi.encodePacked(paraHead, paraHeadSibling));
+    }
+
+    // Verifies if an MMR leaf is contained in an MMR
+    function verifyLeafInMMR(
+        bytes32 leaf,
+        bytes32[] memory orderedBranch,
+        bytes32 peakParaHeadsRoot
+    ) public pure returns(bool) {
+        bytes32 currHash = leaf;
+        for(uint i = 0; i < orderedBranch.length; i++) {
+            currHash = keccak256(abi.encodePacked(currHash, orderedBranch[i]));
```

Comment:
Again, the order matters here. We probably have to use a bitfield for this, since the hashes will be ordered by block height, not bitwise.

Reply:
What's the best way to encode the ordering information into the bitfield?

Reply:
That really depends on the reference implementation by Parity. The easiest for now is probably just to use a `bool[]`, do you want to roll with that for now?

Reply:
Yeah, that seems reasonable for now We can research optimizations later on if we need to.